django/db/backends/mysql/validation.py

```diff
@@ -61,4 +61,17 @@ def check_field(self, field, **kwargs):
                     id='mysql.E001',
                 )
             )
+
+        if field.db_index and field_type.lower() in (
+            'tinyblob', 'blob', 'mediumblob', 'longblob', 'tinytext', 'text',
+            'mediumtext', 'longtext', 'json',
+        ):
+            errors.append(
+                checks.Warning(
+                    'A database index on %s is not supported on this database '
```

Comment:
Should I create a warning for each database? I think that: _"**fields.W162**: ``<database>`` does not support a database index on ``<field data type>`` columns."_ is fine.